### PR TITLE
Implement Error, Debug and Display for KafkaError

### PR DESCRIPTION
**Pull request by thijsc**
Implement these traits so it's possible to integrate `KafkaError` with normal Rust error handling and libraries such as `error-chain`.

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -41,8 +40,145 @@ pub enum KafkaError {
...
+    }
+}
+
 impl From<std::ffi::NulError> for KafkaError {
     fn from(err: std::ffi::NulError) -> KafkaError {
         KafkaError::Nul(err)
     }
 }
+
+fn rd_kafka_resp_err_description(err: RDKafkaRespErr) -> &'static str {
```
Comment (fede1024) [inline src/error.rs]: Could we use the `rd_kafka_get_err_descs` function provided by librdkafka?

Comment (thijsc) [inline src/error.rs]: Oh, didn't know that existed. Will take a look.

Comment (thijsc) [inline src/error.rs]: You probably mean `rd_kafka_err2str`?

**Comment by thijsc** [comment]
Changed the source of the error description to `rd_kafka_err2str`.

File: src/error.rs [hunk 2]
```diff
@@ -41,8 +42,72 @@ pub enum KafkaError {
...
+    }
+}
+
 impl From<std::ffi::NulError> for KafkaError {
     fn from(err: std::ffi::NulError) -> KafkaError {
         KafkaError::Nul(err)
     }
 }
+
+fn rd_kafka_resp_err_description(err: RDKafkaRespErr) -> String {
```
Comment (fede1024) [inline src/error.rs]: Since this is not a native rdkafka function, i would suggest renaming it. Something along the lines of `resp_err_description`.

**Comment by thijsc** [comment]
Renamed the method and rebased. Did it on my master by accident, so it includes #14. Will rebase again once that's merged.

**Comment by thijsc** [comment]
Rebased

**Pull request merged**